runtime/lua/vim/version.lua

```diff
@@ -298,8 +301,9 @@ function M.range(spec) -- Adapted from https://github.com/folke/lazy.nvim
 
   local semver = M.parse(version)
   if semver then
-    local from = semver
-    local to = vim.deepcopy(semver, true)
+    local from = semver --- @type Version?
+    local to = vim.deepcopy(semver, true) --- @type Version?
+    ---@diagnostic disable: need-check-nil
```

Comment:
```suggestion
    assert(mods)
```

Reply:
The issue isn't with `mods` but with `from` and `to`. I made them nullable because some if-blocks will assign `nil`, but then that causes code like `to.patch` to warn. I could add the `assert`s before all of those accesses, but I think it might be an overkill and sacrifice readability.

Reply:
I'm not sure how swapping for an `assert` is less readable than `@diagnostic` lines.

Reply:
Oh wait sorry. Just realised this is `disable` and not `disable-next-line`.

I still think in principle I'd prefer the asserts. I don't think it will reduce readability.